distributed/versions.py

```diff
@@ -7,30 +7,40 @@
 import platform
 import struct
 import sys
-from collections.abc import Callable, Iterable
+from collections.abc import Iterable
 from itertools import chain
-from types import ModuleType
 from typing import Any
 
+try:
+    import importlib.metadata
+except ImportError:
+    # Python 3.7 compatibility
+
+    import pkg_resources
+
+    _version = lambda modname: pkg_resources.get_distribution(modname).version
+else:
+    _version = importlib.metadata.version
```

Comment:
I think it's better to use a `sys.version_info` guard here - because tools like pyupgrade and mypy support it. And it saves you writing a comment

```suggestion
if sys.version_info >= (3, 7):
    from importlib.metadata import version as _version
else:
    def _version(distribution_name):
        return pkg_resources.get_distribution(distribution_name).version
```

Reply:
Thanks @graingert, went back and forth on that. Tooling support is a good point.